lib/runtime/addon.js

```diff
@@ -341,4 +340,34 @@ export default class Addon {
     return configModule && (configModule.default || configModule);
   }
 
+  get blueprintNames() {
+    let names = fs.readdirSync(path.join(this.dir, 'blueprints'));
+    return names;
+  }
+
+  get blueprints() {
+    return this.blueprintNames.map((name) => {
+      let blueprintPath = path.join(this.dir, 'blueprints', name);
+      let Print = require(blueprintPath).default;
+      return new Print({ name, addon: this, description: Print.description });
+    });
+  }
+
+  hasBlueprint(name) {
+    const names = this.blueprintNames;
+    const blueprint = names.includes(name);
+
+    return Boolean(blueprint);
+  }
+
+  getBlueprint(name) {
+    try {
+      let blueprintPath = path.join(this.dir, 'blueprints', name);
+      let Blueprint = require(blueprintPath).default;
+      return new Blueprint({ name, addon: this });
```

Comment:
Yea, for now it's not a problem, but I'm imagining down the line that could produce surprising results if some other code attempted to use it. But yea, caching sounds goods